product_category_active/models/product.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2020 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductCategory(models.Model):
+    _inherit = "product.category"
+
+    active = fields.Boolean(
+        string='Active',
+        default=True,
+        help="If unchecked, it will allow you to hide the "
+             "product category without removing it.")
+
+    @api.constrains("active")
+    def _check_archive(self):
+        to_archive = self.filtered(lambda r: not r.active)
+        if self.env['product.template'].search([
+            '|',
+            ('categ_id', 'in', to_archive.ids),
+            ('categ_id', 'child_of', to_archive.ids),
```

Comment:
No need of this double condition. `child_of` operator already includes it. If you put `child_of, [3, 5]`, it means that it will return `3, 5`, or any of their childrens.
```suggestion
            ('categ_id', 'child_of', to_archive.ids),
```